tests/integration/Statistics/Track/WooCommerceOrdersTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+namespace MailPoet\Test\Statistics\Track;
+
+use DateTime;
+use MailPoet\Models\Newsletter;
+use MailPoet\Models\NewsletterLink;
+use MailPoet\Models\SendingQueue;
+use MailPoet\Models\StatisticsClicks;
+use MailPoet\Models\StatisticsWooCommerceOrders;
+use MailPoet\Models\Subscriber;
+use MailPoet\Statistics\Track\WooCommerceOrders;
+use MailPoet\Tasks\Sending;
+use MailPoet\WooCommerce\Helper;
+use PHPUnit_Framework_MockObject_MockObject;
+use WC_Order;
+
+class WooCommerceOrdersTest extends \MailPoetTest {
+  /** @var Subscriber */
+  private $subscriber;
+
+  /** @var Subscriber */
+  private $newsletter;
+
+  /** @var Sending */
+  private $queue;
+
+  /** @var NewsletterLink */
+  private $link;
+
+  function _before() {
+    parent::_before();
+
+    $this->subscriber = $this->createSubscriber('test@example.com');
+    $this->newsletter = $this->createNewsletter();
+    $this->queue = $this->createQueue($this->newsletter, $this->subscriber);
+    $this->link = $this->createLink($this->newsletter, $this->queue);
+  }
+
+  function testItTracksPayment() {
+    $click = $this->createClick($this->link, $this->subscriber);
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    $order_stats = StatisticsWooCommerceOrders::findMany();
+    expect(count($order_stats))->equals(1);
+    expect($order_stats[0]->newsletter_id)->equals($this->newsletter->id);
+    expect($order_stats[0]->subscriber_id)->equals($this->subscriber->id);
+    expect($order_stats[0]->queue_id)->equals($this->queue->id);
+    expect($order_stats[0]->click_id)->equals($click->id);
+    expect($order_stats[0]->order_id)->equals($order_mock->get_id());
+    expect($order_stats[0]->order_currency)->equals($order_mock->get_currency());
+    expect($order_stats[0]->order_price_total)->equals($order_mock->get_total());
+  }
+
+  function testItTracksPaymentForMultipleNewsletters() {
+    $this->createClick($this->link, $this->subscriber);
+
+    // create click in other newsletter
+    $newsletter = $this->createNewsletter();
+    $queue = $this->createQueue($newsletter, $this->subscriber);
+    $link = $this->createLink($newsletter, $queue);
+    $this->createClick($link, $this->subscriber, 1);
+
+    $order_mock = $this->createOrderMock($this->subscriber->email);
+    $woocommerce_orders = new WooCommerceOrders($this->createWooCommerceHelperMock($order_mock));
+    $woocommerce_orders->trackPaidOrder($order_mock->get_id());
+    expect(count(StatisticsWooCommerceOrders::findMany()))->equals(2);
```

Comment:
Shouldn't this test be a little bit more strict and also test the clicks are created correctly for the correct newsletters?